UnityProject/Assets/Scripts/Player/IPlayerPossessable.cs

```diff
@@ -1,19 +1,185 @@
 using System.Collections;
 using System.Collections.Generic;
+using Core.Utils;
 using HealthV2;
+using Messages.Server;
+using Mirror;
+using Systems.Ai;
 using UnityEngine;
 
 public interface IPlayerPossessable
 {
 
-	LivingHealthMasterBase LHMB { get; set; }
+	public GameObject GameObject { get; }
+	public IPlayerPossessable Possessing { get; set; }
 
-	virtual LivingHealthMasterBase LivingHealthMasterBase()
+	public GameObject PossessingObject { get; set; }
+
+	public Mind PossessingMind { get; set; }
+
+	public IPlayerPossessable PossessedBy { get; set; }
+
+	public MindNIPossessingEvent OnPossessedBy   { get; set; }
+
+	public void InternalOnEnterPlayerControl(GameObject PreviouslyControlling, Mind mind, bool IsServer)
```

Comment:
this function is too big, split it into smaller functions

Reply:
I think it's fine

Reply:
No, split it.

Reply:
split it into what?

Reply:
like I can put it on the player script but it's better to be here more generic

Reply:
> split it into what?

Smaller functions